src/nvim/CMakeLists.txt

```diff
@@ -81,6 +81,8 @@ foreach(subdir
         event
         eval
         lua
+        viml
```

Comment:
what is the logic behind `eval` / `viml` split? Is `viml` for "abstract" parsing/manipulation etc while `eval` is for actual evaluation/execution? Or is it just "old" vs "new" code? Is it important enough to warrant the split?

Reply:
`eval` was intended for expression evaluation only. `viml` is for the whole VimL, including Ex commands. The “expression evaluation” parts like parsing and execution are going to move to viml/ for sure, the core problem is that after the whole VimL is ported to a new parser, the difference between expression evaluation and Ex commands evaluation should be gone (both will use a new VM, whatever it will actually be).

What is left of eval is

1. `typval_T` and related values manipulations. It is more logical to move that near the VM, to viml/ as well (just eval/typval\* to viml/typval\* when viml/ will actually contain some executor, not much hassle).
2. A big bunch of `f_…` functions. My idea is that most logical choice for that is treating them just like `ex_…` functions used for Ex commands: scattered around the codebase, near the domain these functions belong to. But this is too heavy refactoring so probably somebody will move them to a separate file (as a part of #5081) where that will live from now on. File may be moved later.
3. Utilities for things mentioned above: currently existing executing parser, functions, etc.

So it is just “old” vs “new” code, with “new” not using “eval” because it is not about only the expressions. I do not want to have expressions parsing in `eval/` because it is going to be fairly isolated from what already is there, but sharing some things with Ex commands parser (for which there is no preexisting directory in any case).